### ISSUE TITLE
Cannot link pods against ios and osx

### ISSUE THREAD

**Issue by damouse** [CocoaPods/CocoaPods#4478]
Ive tried every combination of `platform`, `link_with`, and `exclusive` in trying to get cocoapods to link with ios and osx targets in one workspace. Seperately I have two workspaces working with the pod with OSX and iOS targets, respectively.

Almost all of the combos get the iOS target to build and run. A few gets the OSX to build, but it fails to link to frameworks at runtime. Here's the closest I've gotten: 

```
use_frameworks!

target :Dee, :exclusive => true do
  platform :ios, '9.0'
  link_with 'Dee'
  pod 'Riffle'
end

target :Doo, :exclusive => true do
  platform :osx, '10.10'
  link_with 'Doo'
  pod 'Riffle'
end
```

Error given at runtime: 
```
dyld: Library not loaded: @rpath/CocoaAsyncSocket.framework/Versions/A/CocoaAsyncSocket
  Referenced from: /Users/Damouse/Library/Developer/Xcode/DerivedData/Dee-bpzfvhkaigsdcgdpgwuuqfkxufgl/Build/Products/Debug/Doo
  Reason: image not found
```

IIRC, almost all of the working configs warn about "... member in multiple groups" as shown below. I have also tried making subspecs named `ios` and `osx` as a long shot to no avail. I think the "multiple groups" error is the problem.

![Multiple groups](http://i.imgur.com/7Ld6D1D.png)

Note: can't get the pod to install with Xcode 7.1. Had to downgrade to 7.0.1-- I've seen the issue posted here, I can link to it if need be. I also have a list of most of the Podfile configs I've tried so far, but most of them either fail earlier than the one above or end with the same linking error. 

[Here's](https://github.com/damouse/TwoPlatformsPlease) a link to a sample repository. The pod is written in swift and has objc dependencies.

**Comment by damouse** [comment]
Updated with log dump screenshot.

**Comment by segiddins** [comment]
That's just a warning, not an error -- it's likely unrelated.

**Comment by TGOS** [comment]
The warning is not unrelated, it gives a good hint what the actual problem is. 

Both, `Dee` and `Doo` want to be linked with the framework `Riffle` (`Riffle` is built as a framework because of `use_frameworks!`). That means this framework is actually built twice (as two different targets), once for the target `Dee` and once for the target `Doo`. But in both cases, the output of the target will create a framework named `Riffle.framework` (or whatever the module_name is for that Pod, can be set in the Podspec). All outputs of all targets are written to an Xcode group name `Products` and CP tries to make two identical entries here, both named `Riffle.framework` and both with identical UUID. And that causes the warning.

To get rid of the warning, try to put the following in your Podfile:

    ENV["COCOAPODS_DISABLE_DETERMINISTIC_UUIDS"] = "true"

This will fix the warning but I don't know if it will also fix your liker problem.

BTW, you can get rid of then `link_with` statements in your Podspec. You can read `target :Dee` as: "*When you link with the target named `Dee`, apply the following settings*" and thus this already implies that that CP shall link with the target `Dee`. IMHO `link_with` is only useful if you don't specify all settings per target but use global settings that shall apply to multiple targets at once.

**Comment by segiddins** [comment]
Link with is actually in the process of being removed...

**Comment by damouse** [comment]
I tried the deterministic UUID envvar before. It seems to just silence the warning, doesn't change the resulting product. 

I did, however, get the project to work. Here's the final Podfile:

```
use_frameworks!

target :Dee, :exclusive => true do
  platform :ios, '9.0'
  pod 'Riffle'
end

target :Doo, :exclusive => true do
  platform :osx, '10.10'
  pod 'Riffle'
end
```

,.. but it wasn't the podfile that did it. I noticed the *Doo* target was missing a build script that the *Dee* project had, so I copied it over. 

Under Doo/Build Phases/New Script Script Phase:

```
"${SRCROOT}/Pods/Target Support Files/Pods-Doo/Pods-Doo-frameworks.sh"
```

I changed the name to reflect the different target, and there you go. Both projects build and run.

**Comment by TGOS** [comment]
But you didn't have to create the script itself, did you? The `Pods-Doo-frameworks.sh` was already present and just not used? This sounds like a bug to me. If the script is created and even does something useful when called (makes the whole project work), why would it not be added to the target?

**Comment by damouse** [comment]
No, I did not create the script.

**Comment by segiddins** [comment]
This should be fixed in CocoaPods v1.0.0.beta.4.

**Comment by sebastiancoke** [comment]
I don't know but i just add the same path problem "@rpath/CocoaAsyncSocket.framework/Versions/A/CocoaAsyncSocket"  in the Build Settings-> Runpath search path, remember using the "+". Then it worked :)
